src/test/java/org/itsallcode/openfasttrace/importer/legacytag/TestLegacyTagImporterFactory.java

```diff
@@ -67,61 +63,89 @@ public void setup()
     public void testFactoryWithEmptyPathConfigListSupportsNothing()
     {
-        assertSupportsFile(emptyList(), PATH1, false);
+        assertSupportsFile(config(), PATH1, false);
     }
 
     @Test
     public void testFactorySupportsFile()
     {
-        assertSupportsFile(asList(createConfig(PATH1)), PATH1, true);
+        assertSupportsFile(config(glob(PATH1)), PATH1, true);
+    }
+
+    @Test
+    public void testFactorySupportsFileWithBasePath()
+    {
+        assertSupportsFile(config("base", glob("path")), "base/path", true);
+    }
+
+    @Test
+    public void testFactoryDoesNotSupportFileWithWrongBasePath()
+    {
+        assertSupportsFile(config("base", glob("path")), "base1/path", false);
+    }
+
+    @Test
+    public void testFactoryDoesNotSupportFileWithWrongPath()
+    {
+        assertSupportsFile(config("base", glob("path")), "base/path1", false);
     }
 
     @Test
     public void testFactoryDoesNotSupportsFile()
     {
-        assertSupportsFile(asList(createConfig(PATH1)), PATH2, false);
+        assertSupportsFile(config(glob(PATH1)), PATH2, false);
     }
 
     @Test(expected = ImporterException.class)
     public void testFactoryThrowsExceptionForUnsupportedFile()
     {
-        createImporter(asList(createConfig(PATH1)), Paths.get(PATH2));
+        createImporter(config(glob(PATH1)), Paths.get(PATH2));
     }
 
     @Test
     public void testFactoryCreatesImporterForSupportedFile() throws IOException
     {
         final File tempFile = this.temp.newFile();
-        final Importer importer = createImporter(asList(createConfig(tempFile.getAbsolutePath())),
+        final Importer importer = createImporter(config(glob(tempFile.getAbsolutePath())),
                 tempFile.toPath());
         assertThat(importer, notNullValue());
     }
 
     @Test(expected = ImporterException.class)
     public void testFactoryThrowsExceptionForMissingFile() throws IOException
     {
-        final Importer importer = createImporter(asList(createConfig(PATH1)), Paths.get(PATH1));
+        final Importer importer = createImporter(config(glob(PATH1)), Paths.get(PATH1));
         importer.runImport();
     }
 
-    private void assertSupportsFile(final List<PathConfig> pathConfigs, final String path,
+    private void assertSupportsFile(final LegacyTagImporterConfig config, final String path,
             final boolean expected)
     {
-        assertThat(create(pathConfigs).supportsFile(Paths.get(path)), equalTo(expected));
+        assertThat(create(config).supportsFile(Paths.get(path)), equalTo(expected));
+    }
+
+    private Importer createImporter(final LegacyTagImporterConfig config, final Path path)
+    {
+        return create(config).createImporter(path, StandardCharsets.UTF_8, this.listenerMock);
+    }
+
+    private LegacyTagImporterConfig config(final String basePath, final PathConfig... pathConfigs)
+    {
+        return new LegacyTagImporterConfig(basePath == null ? null : Paths.get(basePath),
```

Comment:
The null values make this peace of code really hard to read. Especially without context.

Reply:
✔️